src/bin/pgcopydb/catalog.c

```diff
@@ -3143,6 +3146,190 @@ catalog_iter_s_table_part_finish(SourceTablePartsIterator *iter)
 }
 
 
+/*
+ * catalog_iter_s_generated_column_init initializes an Interator over our
+ * catalog of GeneratedColumn.
+ */
+bool
+catalog_iter_s_generated_column_init(GeneratedColumnIterator *iter)
+{
+	sqlite3 *db = iter->catalog->db;
+
+	if (db == NULL)
+	{
+		log_error("BUG: Failed to initialize s_table iterator: db is NULL");
+		return false;
+	}
+
+	iter->column = (GeneratedColumn *) calloc(1, sizeof(GeneratedColumn));
+
+	if (iter->column == NULL)
+	{
+		log_error(ALLOCATION_FAILED_ERROR);
+		return false;
+	}
+
+	char *sql =
+		"  select nspname, relname, attname "
+		"    from s_attr a"
+		"      join s_table t on t.oid = a.oid "
+		"   where a.attisgenerated";
```

Comment:
Is there a distinction to be made between GENERATED ALWAYS and GENERATED BY DEFAULT? (which only is for IDENTITY columns, if I read the docs correctly).

Reply:
ping?

Reply:
@dimitri Yes, we need to differentiate between IDENTITY vs normal types.

The implementation works only for GENERATED ALWAYS with EXPRESSION.
i.e. `information_schema.columns.is_generated` must be `ALWAYS`.

For the IDENTITY types, `information_schema.columns.is_generated` is always `NEVER`. INSERT works fine with `IDENTITY` types, but UPDATE would fail.